trezor/trezor-firmware
# Support Electrum signatures in VerifyMessage.

Implements:
- #2100 in T1 and TT.
- #1880 in T1 and TT.
- Generating Electrum-compatible message signatures in trezorctl.
- Removes Bitcoin Cash from T1 Bitcoin-only firmware.

Screenshots for #1880:
![image](https://user-images.githubusercontent.com/42678794/153052711-f07698f6-c489-452a-aa9d-af61654c0c67.png) ![image](https://user-images.githubusercontent.com/42678794/153059278-b6e71412-efae-4f57-ab68-a3ab6de1c3fc.png)

Suite/Connect:
`VerifyMessage` and `EthereumVerifyMessage` will show one more confirmation screen at the end of the workflow, see screenshot, with button request type `success_verify` and code `Other`.

QA:
Check that T1 and TT are able to verify messages signed by Electrum. Test with Legacy, P2SH SegWit and native SegWit addresses.
